### PR TITLE
mempool: abstract description of tx flooding protocol

### PR DESCRIPTION
**Pull request by cason**
Replaces #13.

Rendered: https://github.com/cometbft/knowledge-base/blob/cason/mempool-propagation/protocols/mempool/flooding-protocol.md

### CODE REVIEW

File: protocols/mempool/flooding-protocol.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+propagation graphs is therefore $2*6 - 4 = 8$.
+
+The above equation uses the size of the set `Recv` as parameter.
+However, in most executions and networks a single node, or at least few nodes,
+should start the broadcast of a transaction via `broadcast(tx)`.
+This means that most nodes are in the `Recv` set,
+namely $|Recv| \approx |V|$, from which the equation becomes:
+
+$$
+messages = 2|E| - |V| + |Bcast| \approx 2|E| - |V|
```
Comment (lasarojc) [inline protocols/mempool/flooding-protocol.md:167]: Where is |Bcast| coming from? I feel like I missed a step here.

Comment (cason) [reply protocols/mempool/flooding-protocol.md:167]: $V = Recv \cup Bcast$, so $|Recv|  = |V| - |Bcast|$.

Comment (cason) [reply protocols/mempool/flooding-protocol.md:167]: In fact, this last step is not really needed.

File: protocols/mempool/flooding-protocol.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
```
Comment (cason) [inline protocols/mempool/flooding-protocol.md:167]:
```suggestion
messages \approx 2|E| - |V|
```

**Review by lasarojc — APPROVED** [review]

**Pull request merged**